src/main/java/io/mosip/digitalcard/controller/DigitalCardController.java

```diff
@@ -85,7 +85,7 @@ public ResponseEntity<?> credentialEvent(@RequestBody EventModel eventModel)  {
             additionalAttributes.remove("credential");
             additionalAttributes.remove("protectionKey");
             additionalAttributes.remove("proof");
-            digitalCardServiceImpl.generateDigitalCard(eventModel.getEvent().getData().get("credential").toString(),
+            digitalCardServiceImpl.generateDigitalCard(eventModel.getEvent().getData().containsKey("credential")?eventModel.getEvent().getData().get("credential").toString():null,
```

Comment:
If this value is null, does the code to reach to datashare and taking the data is already there ?

Reply:
there is one more argument to this method where datashare URL is passed, that will take care of pulling the credential from the datshare service